general/cuda.hpp

```diff
@@ -9,9 +9,39 @@
 // terms of the GNU Lesser General Public License (as published by the Free
 // Software Foundation) version 2.1 dated February 1999.
 
-#ifndef MFEM_CU_STUB_HPP
-#define MFEM_CU_STUB_HPP
+#ifndef MFEM_CUDA_HPP
+#define MFEM_CUDA_HPP
 
+// *****************************************************************************
+#ifdef __NVCC__
+#include <cuda.h>
+template <typename BODY> __global__ static
+void cuKernel(const size_t N, BODY body)
+{
+   const size_t k = blockDim.x*blockIdx.x + threadIdx.x;
+   if (k >= N) { return; }
+   body(k);
+}
+template <size_t BLOCK_SZ, typename DBODY>
+void cuWrap(const size_t N, DBODY &&d_body)
+{
+   const size_t gridSize = (N+BLOCK_SZ-1)/BLOCK_SZ;
+   cuKernel<<<gridSize, BLOCK_SZ>>>(N,d_body);
+}
+constexpr static inline bool cuNvcc() { return true; }
```

Comment:
It was more to know if the NVCC compiler was used `NvccCompilerUsed`?

Reply:
How about `usingNvccCompiler()`, mainly because it will usually be used in if statements
```cpp
if (NvccCompilerUsed()) {
}
```
vs
```cpp
if (usingNvccCompiler()) {
}
```
---
The other point is still unresolved I think
```cpp
if (PA()) {
}
```
vs
```cpp
if (usingPA()) {
}
```

Reply:
`usingPA` and `usingNvccCompiler`